src/components/Researcher/ParticipantList/NewPatientDetail.tsx

```diff
@@ -0,0 +1,47 @@
+import React from "react"
+import QRCode from "qrcode.react"
+import { Grid, Tooltip, TextField, Box } from "@material-ui/core"
+import SnackMessage from "../../SnackMessage"
+import LAMP from "lamp-core"
+import { useSnackbar } from "notistack"
+
+const _qrLink = (credID, password) =>
+  window.location.href.split("#")[0] +
+  "#/?a=" +
+  btoa([credID, password, LAMP.Auth._auth.serverAddress].filter((x) => !!x).join(":"))
+
+const { enqueueSnackbar } = useSnackbar()
+
+export const showResult = (id: string) => {
+  enqueueSnackbar(
+    t("Successfully created Participant id. Tap the expand icon on the right to see credentials and details.", {
+      id: id,
+    }),
+    {
+      variant: "success",
+      persist: true,
+      content: (key: string, message: string) => (
+        <SnackMessage id={key} message={message}>
+          <TextField variant="outlined" size="small" label={t("Temporary email address")} value={`${id}@lamp.com`} />
+          <Box style={{ height: 16 }} />
+          <TextField variant="outlined" size="small" label={t("Temporary password")} value={`${id}`} />
+          <Grid item>
+            <TextField
+              fullWidth
+              label={t("One-time login link")}
+              style={{ marginTop: 16 }}
+              variant="outlined"
+              value={_qrLink(`${id}@lamp.com`, id)}
+              onChange={(event) => {}}
+            />
+            <Tooltip title={t("Scan this QR code on a mobile device to automatically open a user dashboard.")}>
+              <Grid container justify="center" style={{ padding: 16 }}>
+                <QRCode size={256} level="H" value={_qrLink(`${id}@lamp.com`, id)} />
+              </Grid>
+            </Tooltip>
+          </Grid>
+        </SnackMessage>
+      ),
+    }
+  )
+}
```

Comment:
Please make sure the other usage of this (in AddUser.tsx) is also deleted and converted to calling this function.

Reply:
@avaidyam It is not working as expected. As I said above, it is showing some error. I can work on this on Monday.